### PR TITLE
Add dry-struct integration via an extension

### PR DESCRIPTION
**Pull request by flash-gordon**
This took me a dozen of minutes /cc @solnic @cored

### CODE REVIEW

**Comment by solnic** [comment]
😍

File: lib/dry/validation/extensions/struct.rb [hunk 1]
```diff
@@ -0,0 +1,32 @@
+require 'dry-struct'
+
+module Dry
+  module Validation
+    class Schema
+      module StructClassBuilder
```
Comment (cored) [inline lib/dry/validation/extensions/struct.rb]: @flash-gordon Why didn't you just created a class name `Struct < Dry::Struct` and at the test level just use that new class ?

Comment (flash-gordon) [inline lib/dry/validation/extensions/struct.rb]: @cored could you please elaborate on this?

Comment (cored) [inline lib/dry/validation/extensions/struct.rb]: @flash-gordon nvm, I need to understand the code better. My initial implementation had something like 
```ruby 
class Dry::Schema::Struct < Dry::Struct 
... 
end 
```

**Pull request merged**